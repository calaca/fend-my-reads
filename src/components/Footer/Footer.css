footer {
  background-color: #f1f1f1;
  padding: 20px 30px;
  text-align: center;
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.15);
  height: 60px;
  position: absolute;
  bottom: 0;
  width: 100%;
}

.footer-text,
.footer-link {
  color: #bdbdbd;
  font-size: 14px;
  font-weight: 200;
}

.footer-text {
  margin: 0;
}

.footer-link {
  text-decoration: none;
  transition: all 0.4s ease-in-out;
}

.footer-link:hover {
  color: #61dafb;
}

.footer-icon {
  height: 18px;
  width: auto;
  transform:rotate(0deg);
  animation: spin 3s linear infinite;
}

@keyframes spin {
  from {
    transform:rotate(0deg);
  }
  to {
    transform:rotate(360deg);
  }
}
