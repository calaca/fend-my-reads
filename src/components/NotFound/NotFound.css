.app-title {
  background-color: #f1f1f1;
  width: 100%;
  height: 80px;
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.15);
  color: #9c9c9c;
  text-align: center;
  font-size: 36px;
  font-weight: 400;
  text-transform: uppercase;
  margin: 0;
  padding-top: 20px;
}

.not-found-text {
  margin: 70px auto;
  background-color: #fff;
  width: 500px;
  padding: 30px;
  border-radius: 4px;
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.1);
  text-transform: uppercase;
  color: #c0392b;
}

.not-found-text .back-link {
  display: inline-block;
  width: 100%;
  top: 0;
  left: 0;
  padding-bottom: 10px;
  margin-bottom: 10px;
  border-bottom: 1px solid #f1f1f1;
  font-size: 22px;
}

.not-found-text .small,
.not-found-text .big {
  display: inline-block;
  width: 100%;
  text-align: center;
}

.small {
  font-size: 42px;
  font-weight: 200;
}

.big {
  font-size: 72px;
  font-weight: 600;
  color: #c0392b;
}
