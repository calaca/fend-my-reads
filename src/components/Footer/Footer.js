import React from 'react';
import reactIcon from '../../icons/react-icon.svg';
import './Footer.css';

const Footer = () => {
  return (
    <footer>
      <p className="footer-text">made with <img className="footer-icon" src={reactIcon} alt="React" title="React" /> by <a className="footer-link" href="https://calaca.github.io">calaca</a></p>
    </footer>
  )
};

export default Footer;
