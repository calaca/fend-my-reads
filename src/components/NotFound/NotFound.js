import React from 'react';
import { Link } from 'react-router-dom';
import arrow from '../../icons/right-arrow.svg'
import './NotFound.css';

const NotFound = () => {
  return (
    <div className="not-found">
      <h1 className="app-title">My Reads</h1>
      <div className="not-found-text">
        <Link className="back-link" to="/">
          <img src={arrow} alt="Go back" title="Go back" className="back-icon"/>
        </Link>
        <span className="small">Error</span>
        <span className="big">404</span>
        <span className="small">page not found</span>
      </div>
    </div>
  )
};

export default NotFound;
