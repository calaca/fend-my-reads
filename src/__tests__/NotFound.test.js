import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NotFound from '../components/NotFound/NotFound';
import Footer from '../components/Footer/Footer';

describe('<NotFound />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NotFound />);
  });

  it('renders without crashing', () => {
    wrapper;
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders `.not-found-text`', () => {
    expect(wrapper.find('.not-found-text')).toHaveLength(1);
  });

  it('has a link back to home', () => {
    expect(wrapper.find('.back-link').prop('to')).toEqual('/');
  });

  it('renders a <Footer />', () => {
    expect(wrapper.find(Footer)).toHaveLength(1);
  });
});
