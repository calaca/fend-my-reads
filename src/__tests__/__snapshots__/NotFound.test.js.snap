// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="app-title"
>
          My Reads
</h1>,
        <div
          className="not-found-text"
>
          <Link
                    className="back-link"
                    replace={false}
                    to="/"
          >
                    <img
                              alt="Go back"
                              className="back-icon"
                              src="right-arrow.svg"
                              title="Go back"
                    />
          </Link>
          <span
                    className="small"
          >
                    Error
          </span>
          <span
                    className="big"
          >
                    404
          </span>
          <span
                    className="small"
          >
                    page not found
          </span>
</div>,
        <Footer />,
      ],
      "className": "not-found",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Reads",
          "className": "app-title",
        },
        "ref": null,
        "rendered": "My Reads",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="back-link"
              replace={false}
              to="/"
>
              <img
                            alt="Go back"
                            className="back-icon"
                            src="right-arrow.svg"
                            title="Go back"
              />
</Link>,
            <span
              className="small"
>
              Error
</span>,
            <span
              className="big"
>
              404
</span>,
            <span
              className="small"
>
              page not found
</span>,
          ],
          "className": "not-found-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Go back"
                className="back-icon"
                src="right-arrow.svg"
                title="Go back"
/>,
              "className": "back-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Go back",
                "className": "back-icon",
                "src": "right-arrow.svg",
                "title": "Go back",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Error",
              "className": "small",
            },
            "ref": null,
            "rendered": "Error",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "404",
              "className": "big",
            },
            "ref": null,
            "rendered": "404",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "page not found",
              "className": "small",
            },
            "ref": null,
            "rendered": "page not found",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="app-title"
>
            My Reads
</h1>,
          <div
            className="not-found-text"
>
            <Link
                        className="back-link"
                        replace={false}
                        to="/"
            >
                        <img
                                    alt="Go back"
                                    className="back-icon"
                                    src="right-arrow.svg"
                                    title="Go back"
                        />
            </Link>
            <span
                        className="small"
            >
                        Error
            </span>
            <span
                        className="big"
            >
                        404
            </span>
            <span
                        className="small"
            >
                        page not found
            </span>
</div>,
          <Footer />,
        ],
        "className": "not-found",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Reads",
            "className": "app-title",
          },
          "ref": null,
          "rendered": "My Reads",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="back-link"
                replace={false}
                to="/"
>
                <img
                                alt="Go back"
                                className="back-icon"
                                src="right-arrow.svg"
                                title="Go back"
                />
</Link>,
              <span
                className="small"
>
                Error
</span>,
              <span
                className="big"
>
                404
</span>,
              <span
                className="small"
>
                page not found
</span>,
            ],
            "className": "not-found-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="Go back"
                  className="back-icon"
                  src="right-arrow.svg"
                  title="Go back"
/>,
                "className": "back-link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Go back",
                  "className": "back-icon",
                  "src": "right-arrow.svg",
                  "title": "Go back",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error",
                "className": "small",
              },
              "ref": null,
              "rendered": "Error",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "404",
                "className": "big",
              },
              "ref": null,
              "rendered": "404",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "page not found",
                "className": "small",
              },
              "ref": null,
              "rendered": "page not found",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
